### ISSUE TITLE
对LightSANs模型中的多头注意力计算存在疑问

### ISSUE THREAD

**Issue by zyx1017** [RUCAIBox/RecBole#1806]
大佬，对不起，我有个疑问：
在计算多头注意力得分时，为什么不需要像以前的做法一样，需要把之后的交互给mask掉？
比如以往的做法都会加上一个mask矩阵：attention_scores = attention_scores + attention_mask
而本代码见如下，并没有进行mask操作。对此感到疑问！希望能得到解答！！！
attention_scores = torch.matmul(query_layer, key_layer.transpose(-1, -2))
attention_scores = attention_scores / math.sqrt(self.attention_head_size)

\# normalize the attention scores to probabilities.
attention_probs = nn.Softmax(dim=-2)(attention_scores)
attention_probs = self.attn_dropout(attention_probs)
context_layer_item = torch.matmul(attention_probs, value_layer)

**Comment by Sherry-XLL** [comment]
@zyx1017 您好，主要是因为 LightSANs 在经过 item2interest 的转换之后，`attention_scores` 的维度和 `attention_mask` 并不匹配（物品列表的长度降到了 k 维），因此无法直接通过相加进行掩码的 mask 操作。
